frontend/javascripts/admin/job/job_list_view.js

```diff
@@ -0,0 +1,162 @@
+// @flow
+import { PropTypes } from "@scalableminds/prop-types";
+import { Link, type RouterHistory, withRouter } from "react-router-dom";
+import { Table, Spin, Input, Icon } from "antd";
+import { connect } from "react-redux";
+import * as React from "react";
+
+import type { APIJob, APIUser } from "admin/api_flow_types";
+import { getJobs } from "admin/admin_rest_api";
+import Persistence from "libs/persistence";
+import * as Utils from "libs/utils";
+import type { OxalisState } from "oxalis/store";
+
+const { Column } = Table;
+const { Search } = Input;
+
+const typeHint: APIJob[] = [];
+
+type StateProps = {|
+  activeUser: ?APIUser,
+|};
+type Props = { ...StateProps, history: RouterHistory };
+
+type State = {
+  isLoading: boolean,
+  jobs: Array<APIJob>,
+  searchQuery: string,
+};
+
+const persistence: Persistence<State> = new Persistence(
+  { searchQuery: PropTypes.string },
+  "jobList",
+);
+
+class JobListView extends React.PureComponent<Props, State> {
+  intervalID;
+
+  state = {
+    isLoading: true,
+    jobs: [],
+    searchQuery: "",
+  };
+
+  componentWillMount() {
+    this.setState(persistence.load(this.props.history));
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentWillUpdate(nextProps, nextState) {
+    persistence.persist(this.props.history, nextState);
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.intervalID);
+  }
+
+  async fetchData(): Promise<void> {
+    const jobs = await getJobs();
+    this.setState(
+      {
+        isLoading: false,
+        jobs: jobs || [],
```

Comment:
In which case is `|| []` necessary? Shouldn't `getJobs` always yield an array?

Reply:
Oh, yes. It's not necessary at all!